packages/medusa/src/migrations/1692953518123-drop_money_amount_constraints_for_pricing_module.ts

```diff
@@ -0,0 +1,45 @@
+import { MigrationInterface, QueryRunner } from "typeorm"
+
+export class dropMoneyAmountConstraintsForPricingModule1692953518123
+  implements MigrationInterface
+{
+  name = "dropMoneyAmountConstraintsForPricingModule1692953518123"
+
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.query(
+      `
+        CREATE TABLE IF NOT EXISTS "money_amount_variant"
+        (
+            "money_amount_id" text NOT NULL,
+            "variant_id" text NOT NULL
+        );
+
+        INSERT INTO "money_amount_variant" ("money_amount_id", "variant_id")
+          SELECT "id", "variant_id" FROM "money_amount";
+
+        ALTER TABLE "money_amount" DROP COLUMN IF EXISTS "variant_id";
+        CREATE UNIQUE INDEX IF NOT EXISTS "idx_money_amount_variant_money_amount_id_unique" ON "money_amount_variant" ("money_amount_id");
```

Comment:
question: Is this correct? I thought that variants can have many money amounts

Reply:
Multiple Money Amounts should indeed be able to be associated with the same variant - e.g. for price lists.

Reply:
yes, but one money amount should only have one variant. Might have mis-named the index a bit, but the index should be correct right?